models/action.go

```diff
@@ -470,9 +470,17 @@ func UpdateIssuesCommit(doer *User, repo *Repository, commits []*PushCommit) err
 				continue
 			}
 
-			if err = issue.ChangeStatus(doer, repo, true); err != nil {
+			// Check for dependencies, if there aren't any, close it
+			noDeps, err := IssueNoDependenciesLeft(issue)
+			if err != nil {
 				return err
 			}
+
+			if noDeps {
+				if err = issue.ChangeStatus(doer, repo, true); err != nil {
+					return err
+				}
+			}
```

Comment:
I prefer add a else here to notify the UI to say it cannot be closed since it has some dependencies not closed. @kolaente @bkcsoft

Reply:
But it shows an error message if it cannot be closed because there are open dependencies 🤔

Reply:
Like so
![bildschirmfoto vom 2018-01-01 14-38-04](https://user-images.githubusercontent.com/13721712/34468083-694a8bc8-ef01-11e7-8a32-109160a7fd7d.png)

Reply:
@kolaente Yes, I thinks that's expected? Shouldn't that?